subo/command/docs.go

```diff
@@ -0,0 +1,484 @@
+						if err != nil {
+							return errors.Wrap(err, "failed to getCodeText")
+						}
+
+						goSnippets[exampleKey] = text
+					}
+
+					// Add all examples option for function
+					exampleKey := getExampleKey(pkgName, funcNode.Name, "")
+					text, err := funcData.getCodeText(exampleKey, docTemplate, actionTemplate, funcNode.Examples, fset)
+					if err != nil {
+						return errors.Wrap(err, "failed to getCodeText")
+					}
+
+					goSnippets[exampleKey] = text
+				}
+			}
+
+			return nil
+		},
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to Walk")
+	}
+
+	return goSnippets, nil
+}
+
+// getExampleKey returns keys associated to examples
+func getExampleKey(pkgName, funcName, exampleSuffix string) string {
+	key := pkgName
+	if funcName != "" {
+		key = fmt.Sprintf("%s/%s", pkgName, funcName)
+		if exampleSuffix != "" {
+			key = strings.Replace(key, fmt.Sprintf("_%s", exampleSuffix), fmt.Sprintf(":%s", exampleSuffix), 1)
+		}
+	} else if exampleSuffix != "" {
+		key = fmt.Sprintf("%s:%s", key, exampleSuffix)
+	}
+
+	return key
+}
+
+// getCodeText returns code snippets generated from example asts
+func (c *codeData) getCodeText(exampleKey, docTemplate, actionTemplate string, examples []*doc.Example, fset *token.FileSet) (string, error) {
+	err := c.getExampleData(examples, fset)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to getExampleData")
+	}
+
+	// Generate user template action for reinsertion
+	tmpl, err := template.New("NestedTemplateAction").Delims(`[[`, `]]`).Parse(actionTemplate)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to Parse")
+	}
+
+	var buffer bytes.Buffer
+	if err := tmpl.Execute(&buffer, map[string]string{"ExampleKey": exampleKey}); err != nil {
+		return "", errors.Wrap(err, "failed to Execute")
+	}
+
+	c.Action = buffer.String()
+	// Generate code snippets based on `docTemplate` templates
+	tmpl, err = template.New("CodeText").Parse(docTemplate)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to Parse")
+	}
+
+	buffer.Reset()
+	if err := tmpl.Execute(&buffer, c); err != nil {
+		return "", errors.Wrap(err, "failed to Execute")
+	}
+
+	return buffer.String(), nil
+}
+
+// getExampleData loads example asts to codeData
+func (c *codeData) getExampleData(examples []*doc.Example, fset *token.FileSet) error {
+	c.Examples = nil
+	for i, example := range examples {
+		// Get example code snippets ast
+		var buffer bytes.Buffer
+		switch n := example.Code.(type) {
+		case *ast.BlockStmt:
+			for _, n := range n.List {
+				if err := printer.Fprint(&buffer, fset, n); err != nil {
+					return errors.Wrap(err, "failed to Fprint")
+				}
+
+				fmt.Fprint(&buffer, "\n")
+			}
+		}
+
+		// Get example code snippets metadata
+		c.Examples = append(c.Examples,
+			&exampleData{
+				Suffix: example.Suffix,
+				Doc:    example.Doc,
+				Code:   fmt.Sprintf("```go\n%s\n```", buffer.String()),
```

Comment:
```suggestion
				Code:   fmt.Sprintf("```go\n%s```", buffer.String()),
```

Ah-ha! This is the newline I was talking about.

Reply:
> lol, whops I forgot about that newline!

Looks like I stashed this changed locally some 🙃